contracts/USM.sol

```diff
@@ -74,31 +84,41 @@ contract USM is ERC20 {
      * @notice Funds the pool with ETH, minting FUM at its current price and considering if the debt ratio goes from under to over
      * @param ethIn Amount of wrapped Ether to use for minting FUM.
      */
-    function fund(uint256 ethIn) external {
-        require(eth.transferFrom(msg.sender, address(this), ethIn), "Eth transfer fail");
+    function fund(address from, address to, uint ethIn)
+        external
+        onlyHolderOrDelegate(from, "Only holder or delegate")
+        returns (uint)
+    {
+        require(eth.transferFrom(from, address(this), ethIn), "Eth transfer fail");
         require(ethIn > MIN_ETH_AMOUNT, "0.001 ETH minimum"); // TODO: Needed?
         if(debtRatio() > MAX_DEBT_RATIO){
             // calculate the ETH needed to bring debt ratio to suitable levels
             uint ethNeeded = usmToEth(totalSupply()).wadDiv(MAX_DEBT_RATIO).sub(ethPool()).add(1); //+ 1 to tip it over the edge
+            uint fumOut;
             if (ethIn >= ethNeeded) { // Split into two fundings at different prices
-                _fund(msg.sender, ethNeeded);
-                _fund(msg.sender, ethIn.sub(ethNeeded));
-                return;
+                fumOut = _fund(to, ethNeeded);
```

Comment:
Probably simplify 97-99 to:

```
            if (ethIn >= ethNeeded) { // Split into two fundings at different prices
                uint fumOut = _fund(to, ethNeeded);
```

Reply:
Simplified to a one-liner, hopefully readable enough.